libr/arch/p/or1k/plugin.c

```diff
@@ -16,8 +16,22 @@ struct operands {
 };
 
 // XXX remove globals and move into init/fini data pointer or just reimplement this into an ctual anal plugin for 5.9 so maybe its not worth
-static R_TH_LOCAL ut32 cpu[32] = {0}; /* register contents */
-static R_TH_LOCAL ut32 cpu_enable; /* allows to treat only registers with known value as valid */
+struct or1k_regs {
+	ut32 cpu[32]; /* register contents */
+	ut32 cpu_enable; /* allows to treat only registers with known value as valid */
+};
+
+static bool or1k_init(RArchSession *s) {
+	r_return_val_if_fail (!s->user, false);
```

Comment:
imho its better to use `->data` instead.. but im seeing some inconsistencies (wasm plugin is >user) but evm and bpf use >data instead.

imho user holds a user controlled pointer. and data its plugin-specific data. the thing is that there's no comment to clarify this, so its normal that is missused, and also i see no way to set this user pointer from the arch api.

```
130$ git grep '>user' *.c
arch_session.c:         ai->user = NULL;
0$ git grep '>data' *.c
arch_session.c:                 bool res = init (ai); // must fill ai->data
0$
```

this is the only comment referencing this. so imho we can add comments about this in the r_arch.h and so

Reply:
ack, I didn't knew the difference, I am more used to drivers using a pointer called `priv`.